validator/cycle/process_tasks.py

```diff
@@ -367,12 +367,47 @@ async def move_tasks_to_preevaluation_loop(config: Config):
 
 
 async def evaluate_tasks_loop(config: Config):
+    gpu_ids = [i for i in range(len(cst.GPU_IDS))]
+
+    # Create queues for tasks and GPUs
+    task_queue = asyncio.Queue()
+    gpu_queue = asyncio.Queue()
+    for gpu_id in gpu_ids:
+        await gpu_queue.put(gpu_id)
+
+    async def evaluation_worker():
+        while True:
+            try:
+                task = await asyncio.wait_for(task_queue.get(), timeout=1)
+                gpu_id = await gpu_queue.get()
+
+                try:
+                    await _evaluate_task(task, [gpu_id], config)
+                finally:
+                    # Return GPU to pool
+                    await gpu_queue.put(gpu_id)
+                    task_queue.task_done()
+            except asyncio.TimeoutError:
+                # No more tasks in queue
+                break
+
     while True:
         tasks_to_evaluate = await tasks_sql.get_tasks_with_status(TaskStatus.PREEVALUATION, psql_db=config.psql_db)
         if tasks_to_evaluate:
             logger.info(f"There are {len(tasks_to_evaluate)} tasks awaiting evaluation")
             for task in tasks_to_evaluate:
-                await _evaluate_task(task, config)
+                await task_queue.put(task)
+
+            # Start workers (one per GPU)
+            workers = [asyncio.create_task(evaluation_worker()) for _ in gpu_ids]
```

Comment:
I think this queuing approach defeats the purpose of having a queue:


Gets ALL tasks upfront and puts them in a queue
Waits for ALL tasks to complete before getting more